### DISCUSSION TITLE
How to run vcpkg in a clean shell environment

### DISCUSSION THREAD

**Discussion by kmrashad** [microsoft/vcpkg#24027] [Q&A] [answered]
Using `vcpkg env`,  we can build packages in a special clean environment that is isolated from the current command prompt to ensure build reliability and consistency.  However, I am running vcpkg in CI machine which updates packages like powershell and this affects triplet-api of all packages. So the binary cache becomes invalid if there is a different version of powershell than the one used when creating the cache archive.

How can I use `vcpkg env` to get a clean environment before running `vcpkg install` ?

Is there any option or environment variable to ensure a clean shell when running vcpkg ?

**Comment by Neumann-A** [comment, marked as answer]
the option `--x-abi-tools-use-exact-versions` will force the version used by vcpkg.

**Reply by kmrashad** [reply]
Thanks a lot!. It worked as expected.
